iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp

```diff
@@ -217,22 +226,34 @@ class ConvertFuncWithHALInterface : public ConvertToLLVMPattern {
     }
 
     TypeConverter::SignatureConversion signatureConverter(/*numOrigInputs=*/0);
-
-    // func foo(%packed_buffer_args: !llvm<i8**>, %push_constant: !llvm<i32*>)
+    // clang-format off
+    // func foo(%packed_buffer_args: !llvm<i8**>, %push_constant: !llvm<i32*>, thread_idx: i32, thread_idy, thread_idz: i32)
+    // clang-format on
     MLIRContext *context = rewriter.getContext();
     auto packedBuffersArgsTy =
         LLVM::LLVMType::getInt8PtrTy(context).getPointerTo();
     auto pushConstantArgTy = LLVM::LLVMType::getInt32Ty(context).getPointerTo();
+    auto threadIdXTy = LLVM::LLVMType::getInt32Ty(context);
+    auto threadIdYTy = LLVM::LLVMType::getInt32Ty(context);
+    auto threadIdZTy = LLVM::LLVMType::getInt32Ty(context);
     signatureConverter.addInputs(packedBuffersArgsTy);
     signatureConverter.addInputs(pushConstantArgTy);
+    signatureConverter.addInputs(threadIdXTy);
+    signatureConverter.addInputs(threadIdYTy);
+    signatureConverter.addInputs(threadIdZTy);
 
     // Create the new function's signature.
     Location loc = funcOp.getLoc();
     auto newFuncOp = rewriter.create<FuncOp>(
         loc, funcOp.getName(),
         rewriter.getFunctionType(signatureConverter.getConvertedTypes(),
-                                 llvm::None),
-        ArrayRef<NamedAttribute>());
+                                 llvm::None));
+
+    // TODO(ataei): Bettery way of handling WorkgroupFunAttr.
```

Comment:
Wondering what would be the be the better way?

Reply:
Passing all attributes at the construction didn't work. added a better comment